File: resources/documentation/test-automation/selenium-webdriver-tutorial/scripts/java/LoginPage.java
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class LoginPage {
  private final WebDriver driver;
  private final WebDriverWait wait;

  private final By emailInput = By.id("email");
  private final By passwordInput = By.id("password");
  private final By loginButton = By.cssSelector("[data-testid='login-button']");
  private final By errorMessage = By.id("error");

  public LoginPage(WebDriver driver) {
    this.driver = driver;
    this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
  }

  public void enterEmail(String email) {
    driver.findElement(emailInput).sendKeys(email);
  }

  public void enterPassword(String password) {
    driver.findElement(passwordInput).sendKeys(password);
  }

  public void clickLogin() {
    driver.findElement(loginButton).click();
  }

  public String getErrorMessage() {
    return wait.until(
      ExpectedConditions.visibilityOfElementLocated(errorMessage)
    ).getText();
  }
}

File: resources/documentation/test-automation/bdd-testing-with-cucumber/pages/LoginPage.js
// pages/LoginPage.js
class LoginPage {
  constructor(page) {
    this.page = page;
    this.emailInput = page.locator('[data-testid="email-input"]');
    this.passwordInput = page.locator('[data-testid="password-input"]');
    this.submitButton = page.locator('[data-testid="submit-button"]');
  }

  async goto() {
    await this.page.goto('/login');
  }

  async enterEmail(email) {
    await this.emailInput.fill(email);
  }

  async enterPassword(password) {
    await this.passwordInput.fill(password);
  }

  async clickButton(buttonName) {
    if (buttonName === 'Sign In') {
      await this.submitButton.click();
    }
  }

  async login(email, password, buttonName) {
    await this.goto();
    await this.enterEmail(email);
    await this.enterPassword(password);
    await this.clickButton(buttonName);
  }
}

module.exports = { LoginPage };
